Sources/SmileID/Classes/DocumentVerification/Model/OrchestratedDocumentVerificationViewModel.swift

```diff
@@ -103,125 +102,96 @@ internal class IOrchestratedDocumentVerificationViewModel<T, U: JobResult>: Obse
-                documentFront: frontDocumentUrl,
-                documentBack: backDocumentUrl,
-                selfie: selfieFile,
-                livenessImages: livenessFiles ?? []
-            )
-        } catch {
-            print("Error saving document images: \(error)")
-            onError(error: SmileIDError.unknown("Error saving document images"))
-            return
-        }
-
-        let authRequest = AuthenticationRequest(
-            jobType: jobType,
-            enrollment: false,
-            jobId: jobId,
-            userId: userId
-        )
-
-        let auth = SmileID.api.authenticate(request: authRequest)
-        networkingSubscriber = auth.flatMap { [self] authResponse in
+                let authRequest = AuthenticationRequest(
+                    jobType: jobType,
+                    enrollment: false,
+                    jobId: jobId,
+                    userId: userId
+                )
+                let authResponse = try await SmileID.api.authenticate(request: authRequest).async()
                 let prepUploadRequest = PrepUploadRequest(
                     partnerParams: authResponse.partnerParams.copy(extras: self.extraPartnerParams),
                     allowNewEnroll: String(allowNewEnroll), // TODO - Fix when Michael changes this to boolean
                     timestamp: authResponse.timestamp,
                     signature: authResponse.signature
                 )
-                return SmileID.api.prepUpload(request: prepUploadRequest)
-            }
-            .flatMap { prepUploadResponse in
-                SmileID.api.upload(zip: zip, to: prepUploadResponse.uploadUrl)
-            }
-            .zip(auth)
-            .flatMap { _, authResponse -> AnyPublisher<JobStatusResponse<U>, Error> in
-                let jobStatusRequest = JobStatusRequest(
-                    userId: authResponse.partnerParams.userId,
-                    jobId: authResponse.partnerParams.jobId,
-                    includeImageLinks: false,
-                    includeHistory: false,
-                    timestamp: authResponse.timestamp,
-                    signature: authResponse.signature
+                let prepUploadResponse = try await SmileID.api.prepUpload(request: prepUploadRequest).async()
+                let _ = try await SmileID.api.upload(
+                    zip: zip,
+                    to: prepUploadResponse.uploadUrl
+                ).async()
+                didSubmitJob = true
+                do {
+                    try LocalStorage.moveToSubmittedJobs(jobId: self.jobId)
+                } catch {
+                    print("Error moving job to submitted directory: \(error)")
+                    self.onError(error: error)
+                }
+                DispatchQueue.main.async { self.step = .processing(.success) }
+            } catch {
+                didSubmitJob = if SmileID.allowOfflineMode { false } else { true } // TODO review this
```

Comment:
We need to set this value based on the type of error we get. If it is a network error, didSubmitJob = false and we return success. if it is some other kind of error, we call onError